### DISCUSSION TITLE
REM Chem not dosing acid with EZO-PMP

### DISCUSSION THREAD

**Discussion by ddyer8** [tagyoureit/nodejs-poolController#562] [Nixie]
My logs appear that njs is trying to dose the pool, but I don't see any other evidence that it's happening.  In the logs I see:

3|njsPC      | [8/3/2022, 07:57:23] info: Chem acid demand calculated 115mL for 4min 36sec Tank Level: 4
3|njsPC      | [8/3/2022, 07:57:23] info: Starting a new acid dose 50mL
3|njsPC      | [8/3/2022, 07:57:23] info: Attempting to dose ezo pump

I have also tried doing a manual does, but I don't see any action taken.

There is an issue that the tank configuration does not appear under the Chemistry->Hardware->pH Pump section if Type = Atlas EZO-PMP.  When trying to set the tank level with my initial configuration, capacity was showing 0.  I had to change type to Relay Pump, set capacity to 5 gal, then return the Type to EZO.
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/9251198/replay.zip)

**Comment by rstrouse** [comment]
I will have a look.  Which EZO-PMP do you have?  There are two versions.

**Comment by ddyer8** [comment]
Must be most recent. I just received it. 

Thank you,
Dustin

**Comment by ddyer8** [comment]
Oh, it’s the smaller one - not the large. 

Thank you,
Dustin

**Comment by ddyer8** [comment]
Also, the pump seems to work as intended when commanded through the REM interface. 

Thank you,
Dustin

**Comment by rstrouse** [comment]
Pull njsPC, REM, and dashPanel.  I normalized the ezo-pmp so that it does not use the internal calculated dosing rate.  As it turns out that rate is only good for dosing volume over time which is no good for our function.  You should be able to calibrate it and supply tank levels just like a relay pump now and it will use the calibrated values that your set up in njsPC.

**Comment by ddyer8** [comment]
Ok. I was able to does the pool manually now. The pool pH is too low to request an automatic dose. 

I am seeing an issue where the pH is not updating on the dashPanel. I don’t think this is new. Seems like it only updates if I make a change to the settings such that requires clicking “Save Controller” button under the REM Chem menu. 

Thank you,
Dustin

**Reply by rstrouse** [reply]
Where are you getting the pH value from?  Is it an Atlas probe or is it coming from somewhere else?

If it is then this comes from a feed on REM and updates whenever there is a pH change.

**Reply by ddyer8** [reply]
Yes, it's from Atlas EZO-PH.  The feed comes from REM, but it did not seem to be updating when changed.  I just un-checked the box for send only when changed.  Now the dashPanel matches immediately after the change. I'll have to keep watching to ensure it stays sync'd.

I do see warnings continually for stopping polling from REM:
1|REM      | warn: Atlas_EZO-HUM Suspend Polling 3
1|REM      | warn: Atlas_EZO-ORP Suspend Polling 3

I think there used to be one for the EZO-PH too.  Is this due to the "change-only feature?  The ORP value does appear to be working properly for "change-only" (dashPanel and REM always have same values).

**Reply by rstrouse** [reply]
Yeah that is simply saying that the I2c bus is busy and is normal.

**Comment by ddyer8** [comment]
Atlas EZO-PH

Thank you,
Dustin
